apps/prairielearn/src/components/QuestionContainer.html.ts

```diff
@@ -0,0 +1,400 @@
+import { html, unsafeHtml } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+
+import { CourseInstance, Issue, Submission, User } from '../lib/db-types.js';
+import { idsEqual } from '../lib/id.js';
+
+import { Modal } from './Modal.html.js';
+
+// Only shows this many recent submissions by default
+const MAX_TOP_RECENTS = 3;
+
+type QuestionContext =
+  | 'student_exam'
+  | 'student_homework'
+  | 'instructor'
+  | 'public'
+  | 'manual_grading';
+
+export function QuestionContainer({
+  resLocals,
+  question_context,
+}: {
+  resLocals: Record<string, any>;
+  question_context: QuestionContext;
+}) {
+  const {
+    question,
+    issues,
+    variant,
+    instance_question,
+    variantToken,
+    __csrf_token,
+    questionJsonBase64,
+    urlPrefix,
+    course_instance,
+    authz_data,
+    authz_result,
+    devMode,
+    is_administrator,
+    showTrueAnswer,
+    showSubmissions,
+    submissions,
+    submissionHtmls,
+    answerHtml,
+  } = resLocals;
+
+  return html`
+    <div
+      id="question-0"
```

Comment:
I couldn't find any reference. I removed it in c8d43cede, I'll assume that if it passes CI it should work. I tested locally with v2, v3 and external grading, in instructor and student mode, and it seems to work without it. I did some git blame work, and it seems to be there since before question.ejs was a partial, and it wasn't in use elsewhere at the time (7 years ago) either.